packages/internals/src/migrateTypes.ts

```diff
@@ -1,5 +1,71 @@
 // eslint-disable-next-line @typescript-eslint/no-namespace
 export namespace MigrateTypes {
+  export type MigrationDirectory = {
+    /**
+     * Relative path to a migration directory from `baseDir`.
+     * E.g., `./20201117144659_test`.
+     */
+    path: string
+    /**
+     * Description of the migration file in the directory.
+     */
+    migrationFile: {
+      /**
+       * Relative path to the migration file from `path`.
+       * E.g., `./migration.sql`.
+       */
+      path: 'migration.sql' | (string & {})
```

Comment:
I'm not sure if it's useful to encode an example in the type (especially if it's already present in the doc comment too). Do we ever have to write a string literal here in the real code outside of tests?

Reply:
(not blocking, just a nit/question)

Reply:
We actually always write a string literal there. Before, it was hardcoded in Rust. Now, it's hardcoded in TypeScript.